internal/proxy/task_query.go

```diff
@@ -543,22 +542,29 @@ func reduceRetrieveResults(ctx context.Context, retrieveResults []*internalpb.Re
 	idSet := make(map[interface{}]struct{})
 	cursors := make([]int64, len(validRetrieveResults))
 
+	realLimit := typeutil.Unlimited
 	if queryParams != nil && queryParams.limit != typeutil.Unlimited {
-		loopEnd = int(queryParams.limit)
-
```

Comment:
if we want reduce stop until best, then we should not set loopEnd to limit which may terminate the loop too early